### PR TITLE
Small libraries:iio fixes

### PR DESCRIPTION
**Pull request by mchindri**
Some commits from #708

### CODE REVIEW

**Comment by mchindri** [comment]
v2:
 - Fix cppcheck

File: projects/iio_demo/src/main.c [hunk 1]
```diff
@@ -201,6 +201,8 @@ int main(void)
 	iio_demo_out_init_par = (struct iio_demo_init_param) {
 		.name = demo_device_output,
 		.num_channels = 4,
+		.dev_global_attr = 1100,
+		.dev_ch_attr = 1111,
```
Comment (cristiantpop) [inline projects/iio_demo/src/main.c:205]: Why this values?

Comment (mchindri) [reply projects/iio_demo/src/main.c:205]: No particular reason, I just wanted them to be different from each other in order to see the difference in iioosciloscope.

**Comment by mchindri** [comment]
v3:
 - Rebase onto master and fix conflicts.

Will merge today if no more comments

**Pull request merged**